pkg/cmd/roachtest/operations/changefeeds/utils.go

```diff
@@ -0,0 +1,384 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package changefeeds
+
+import (
+	"context"
+	gosql "database/sql"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeedbase"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/operation"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/operations/helpers"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/roachtestflags"
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/util/hlc"
+	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
+	"github.com/cockroachdb/cockroach/pkg/util/randutil"
+)
+
+// Constants defining environment variables to manage changefeed jobs.
+const (
+	maxChangefeeds          = "OPS_MAX_CF"             // Max number of changefeeds allowed.
+	maxPctChangeFeedsScanOn = "OPS_MAX_PCT_CF_SCAN_ON" // Max percentage of changefeeds with scan "yes" or "only".
+)
+
+// jobDetails contains the job ID and its current state as obtained from the DB.
+type jobDetails struct {
+	jobID              string                    // Unique ID of the changefeed job.
+	state              jobs.State                // Current state of the job (e.g., running, paused, etc.).
+	payload            *jobspb.ChangefeedDetails // required payload details for the job
+	highWaterTimestamp hlc.Timestamp             // high watermark timestamp
+}
+
+// updatePayloadForJobs fetches additional changefeed payload details for specific jobs from the database
+// and populates the same in the jobdetails and returns
+func updatePayloadForJobs(
```

Comment:
nit: it's not clear from the name or doc comment whether or not this modifies or reuses the passed-in job detailses.